tmt/steps/__init__.py

```diff
@@ -1958,7 +1958,16 @@ def _login(
             # Attempt to push the workdir to the guest
             try:
                 guest.push()
-                cwd = cwd or self.parent.plan.worktree
+                worktree = self.parent.plan.worktree
+                test_path = self.parent.plan.discover.tests(
+                    )[-1].path if self.parent.plan.discover.tests() else None
```

Comment:
I'd recommend calling `self.parent.plan.discover.tests()` just once, it may be costly to collect all tests just to pick the last one.